hotel-api-persistence/src/main/java/com/fponce/hotelapp/persistence/impl/HotelRepositoryImpl.java

```diff
@@ -31,7 +30,13 @@ public void createHotel(UUID id, String name, int category) throws SQLException
         sqlWithParameters.put("name", name);
         sqlWithParameters.put("category", category);
         sqlWithParameters.apply(preparedStatement);
-        preparedStatement.executeUpdate();
+        int rowsUpdated = preparedStatement.executeUpdate();
+        if(rowsUpdated > 0) {
```

Comment:
I don't see any use in  if/else approach. If the Hotel entity was created we return the instance, if not we throw an exception, why is Optional useful here?. Let me know your thoughts

Reply:
I was trying to cover the case in which it return 0 (no rows inserted)  without a SQLException. As I read more about it I see there is no case like that, unless you use other kind of SQL statement mixture.